exp/api/v1alpha3/azuremachinepool_types.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/errors"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+)
+
+type (
+	AzureMachineTemplate struct {
+		// VMSize is the size of the Virtual Machine to build.
+		// See https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#virtualmachinesizetypes
+		VMSize string `json:"vmSize"`
+
+		// Image is used to provide details of an image to use during Virtual Machine creation.
+		// If image details are omitted the image will default the Azure Marketplace "capi" offer,
+		// which is based on Ubuntu.
+		// +kubebuilder:validation:nullable
+		// +optional
+		Image *infrav1.Image `json:"image,omitempty"`
```

Comment:
Does this need to be a pointer?

Reply:
It is in VM and seems like it should follow the same structure.

Should it be this way for either? I didn't want to take an opinion in this PR. I would rather follow the current idiom and then propose the change in a subsequent PR.

Reply:
+1 on staying away from larger opinionated refactors in this feature PR and opting for consistency. Let's not introduce more tech debt where we can though, I think it's okay to diverge from VM implementation when the improvement is trivial and non-controversial (this is a general comment, not for this particular line).

Reply:
As long as we're comfortable changing this later, I'm fine with addressing this later.